scapy/layers/isakmp.py

```diff
@@ -27,96 +39,125 @@
-                                               "XAUTHInitRSAEncryption": 65007,
-                                               "XAUTHRespRSAEncryption": 65008,
-                                               "XAUTHInitRSARevisedEncryption": 65009,  # noqa: E501
-                                               "XAUTHRespRSARevisedEncryptio": 65010, }, 0),  # noqa: E501
-                        "GroupDesc": (4, {"768MODPgr": 1,
-                                          "1024MODPgr": 2,
-                                          "EC2Ngr155": 3,
-                                          "EC2Ngr185": 4,
-                                          "1536MODPgr": 5,
-                                          "2048MODPgr": 14,
-                                          "3072MODPgr": 15,
-                                          "4096MODPgr": 16,
-                                          "6144MODPgr": 17,
-                                          "8192MODPgr": 18, }, 0),
-                        "GroupType": (5, {"MODP": 1,
-                                          "ECP": 2,
-                                          "EC2N": 3}, 0),
-                        "GroupPrime": (6, {}, 1),
-                        "GroupGenerator1": (7, {}, 1),
-                        "GroupGenerator2": (8, {}, 1),
-                        "GroupCurveA": (9, {}, 1),
-                        "GroupCurveB": (10, {}, 1),
-                        "LifeType": (11, {"Seconds": 1,
-                                          "Kilobytes": 2}, 0),
-                        "LifeDuration": (12, {}, 1),
-                        "PRF": (13, {}, 0),
-                        "KeyLength": (14, {}, 0),
-                        "FieldSize": (15, {}, 0),
-                        "GroupOrder": (16, {}, 1),
-                        }
-
-# the name 'ISAKMPTransformTypes' is actually a misnomer (since the table
-# holds info for all ISAKMP Attribute types, not just transforms, but we'll
-# keep it for backwards compatibility... for now at least
-ISAKMPTransformTypes = ISAKMPAttributeTypes
-
-ISAKMPTransformNum = {}
-for n in ISAKMPTransformTypes:
-    val = ISAKMPTransformTypes[n]
-    tmp = {}
-    for e in val[1]:
-        tmp[val[1][e]] = e
-    ISAKMPTransformNum[val[0]] = (n, tmp, val[2])
-del n
-del e
-del tmp
-del val
+# see http://www.iana.org/assignments/ipsec-registry.xml for details
+ISAKMPAttributeTypes = {
+    "Encryption": (1, {"DES-CBC": 1,
+                       "IDEA-CBC": 2,
+                       "Blowfish-CBC": 3,
+                       "RC5-R16-B64-CBC": 4,
+                       "3DES-CBC": 5,
+                       "CAST-CBC": 6,
+                       "AES-CBC": 7,
+                       "CAMELLIA-CBC": 8, }, 0),
+    "Hash": (2, {"MD5": 1,
+                 "SHA": 2,
+                 "Tiger": 3,
+                 "SHA2-256": 4,
+                 "SHA2-384": 5,
+                 "SHA2-512": 6, }, 0),
```

Comment:
Personally, I'd prefer an even more rigorous indentations (newline after every opening parenthesis (brace, ...)  and the closing parenthesis (brace, ...) on a single line, aligned vertically), even more readable, because it aligns the elements on the same level. But YMMV of course...


https://github.com/secdev/scapy/blob/45da0b3ebfa08e7d5f83588811e920fe29310e82/scapy/contrib/ikev2.py#L33-L77

Reply:
I agree but I wanted to keep consistency. One day we'll just black everything...